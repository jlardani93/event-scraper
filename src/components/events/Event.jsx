import React from 'react'

export default function event(props){
  return (
    <div className="gridCell">

      <div className="event">
        <a href={props.eventData.eventUrl}>
          <div>
            <h3>{props.eventData.eventTitle}</h3>
            <img src={props.eventData.imgUrl}/>
            <p>{props.eventData.eventTime.trim()}</p>
            <p>{props.eventData.eventLocation}</p>
          </div>
        </a>
      </div>


      <style jsx>
      {`
        img {
          display: block;
          margin-right: auto;
          margin-left: auto;
          width: 90%;
          height: auto;
          border: 1px solid grey;
          border-radius: 5px;
          box-shadow: 0 0 5px grey inset;
        }

        a {
          text-decoration: none;
          color: black;
        }

        .event {
          height: 95%;
          padding: 10px;
          border: 1px solid grey;
          border-radius: 5px;
          background-color: #bcc6b1;
          y-overflow: hidden;
          cursor: pointer;
          transition: transform 0.1s, box-shadow 0.1s;

          -webkit-animation: fadein 2s;
          -moz-animation: fadein 2s;
          -ms-animation: fadein 2s;
          -o-animation: fadein 2s;
            animation: fadein 2s;
        }

        .event:hover {
          box-shadow: 0 0 3px grey inset;
          transform: translate(0, -10px);
          transition: transform 0.1s, box-shadow 0.1s;
        }

        .gridCell {
          padding: 10px;
        }

        @keyframes fadein {
            from { opacity: 0; }
            to   { opacity: 1; }
        }

        @-moz-keyframes fadein {
            from { opacity: 0; }
            to   { opacity: 1; }
        }

        @-webkit-keyframes fadein {
            from { opacity: 0; }
            to   { opacity: 1; }
        }

        @-ms-keyframes fadein {
            from { opacity: 0; }
            to   { opacity: 1; }
        }
      `}
      </style>
    </div>
  )
}
