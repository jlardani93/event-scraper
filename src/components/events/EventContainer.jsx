import React from 'react'
import Event from './Event'
import EventFilter from './EventFilter'

class eventContainer extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      events: [],
      stanfordEvents: [],
      eventBriteEvents: [],
      meetUpEvents: [],
      showStanford: true,
      showEventBrite: true,
      showMeetUp: true,
      activeButtons: [true, true, true]
    }
    this.onSelectFilter = this.onSelectFilter.bind(this)
  }

  componentWillMount(){
    fetch('/scrape')
      .then( res => res.json())
      .then( res => {
        //Converts nested array into a single, flat array
        const newEventsArray = res.reduce((sum, siteArray) => sum.concat(siteArray), [])
        this.setState({
          events: newEventsArray,
          stanfordEvents: res[0],
          eventBriteEvents: res[1],
          meetUpEvents: res[2],
        })
      })
      .catch( err => {
        // console.log(err)
      })
  }

  updateEvents(){
    let newEvents = [];
    if (this.state.showStanford) newEvents = newEvents.concat(this.state.stanfordEvents)
    if (this.state.showEventBrite) newEvents = newEvents.concat(this.state.eventBriteEvents)
    if (this.state.showMeetUp) newEvents = newEvents.concat(this.state.meetUpEvents)
    this.setState({events: newEvents})
  }

  updateButtons(){
    const newActiveButtons = [
      (this.state.showStanford == true),
      (this.state.showEventBrite == true),
      (this.state.showMeetUp == true)
    ]
    this.setState({activeButtons: newActiveButtons})
  }

  onSelectFilter(property){
    const newValue = !this.state[property]
    const onUpdate = (function(){
      this.updateEvents();
      this.updateButtons();
    }).bind(this)
    this.setState({[property]: newValue}, onUpdate)
  }

  render(){
    return (
      <div>
        <EventFilter handleSelectFilter={this.onSelectFilter} activeButtons={this.state.activeButtons} />
        <div id="eventContainer">
          {this.state.events.map( (event, i) =>
            <Event key={i} eventData={event} />
          )}
        </div>


        <style jsx>
          {`
            #eventContainer {
              display: grid;
              grid-template-columns: repeat(4, 1fr);
              padding-top: 10px;
            }
          `}
        </style>


      </div>
    )
  }
}

export default eventContainer;
