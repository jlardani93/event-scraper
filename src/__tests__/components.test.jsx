import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import EventContainer from './../components/events/EventContainer'
import Event from './../components/events/event'

describe('EventContainer', () => {
  it ('renders without throwing error', () => {
    const div = document.createElement('div');
    ReactDOM.render(<EventContainer />, div);
  })

  it ('renders correctly', () => {
    function componentToTest(){
      return <EventContainer />
    }
    const tree = renderer
      .create(<componentToTest />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})

describe('Event', () => {
  it ('renders without throwing error', () => {
    const div = document.createElement('div');
    ReactDOM.render(<EventContainer />, div);
  })

  it ('renders correctly', () => {
    const mockData = {
      eventTitle: 'title',
      eventTime: 'time',
      imgUrl: 'https://media.istockphoto.com/photos/red-apple-picture-id495878092?k=6&m=495878092&s=612x612&w=0&h=q9k5jN-1giBGZgTM6QhyKkPqtGf6vRpkgDzAwEz9DkY=',
      eventUrl: 'https://www.transparentclassroom.com/'
    }

    function componentToTest(props){
      return <Event eventData={props.mockData}/>
    }
    const tree = renderer
      .create(<componentToTest mockData={mockData}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
