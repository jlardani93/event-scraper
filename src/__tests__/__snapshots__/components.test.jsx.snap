// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event renders correctly 1`] = `
<componentToTest
  mockData={
    Object {
      "eventTime": "time",
      "eventTitle": "title",
      "eventUrl": "https://www.transparentclassroom.com/",
      "imgUrl": "https://media.istockphoto.com/photos/red-apple-picture-id495878092?k=6&m=495878092&s=612x612&w=0&h=q9k5jN-1giBGZgTM6QhyKkPqtGf6vRpkgDzAwEz9DkY=",
    }
  }
/>
`;

exports[`EventContainer renders correctly 1`] = `<componentToTest />`;
